ledger-service/fetch-contracts/src/main/scala/fetchcontracts/domain/ContractTypeId.scala

```diff
@@ -71,17 +71,20 @@ object ResolvedQuery {
   }
 
   def apply(resolved: Set[ContractTypeId.Resolved]): Unsupported \/ ResolvedQuery = {
+    import com.daml.nonempty.{NonEmpty, Singleton}
     val (templateIds, interfaceIds) = partition(resolved)
-    if (templateIds.nonEmpty && interfaceIds.nonEmpty) {
-      -\/(CannotQueryBothTemplateIdsAndInterfaceIds)
-    } else if (templateIds.isEmpty && interfaceIds.size > 1) {
-      -\/(CannotQueryManyInterfaceIds)
-    } else if (templateIds.isEmpty && interfaceIds.size == 1) {
-      \/-(ByInterfaceId(interfaceIds.head))
-    } else if (templateIds.nonEmpty) {
-      \/-(ByTemplateIds(templateIds))
-    } else {
-      -\/(CannotBeEmpty)
+    templateIds match {
+      case NonEmpty(templateIds) =>
+        interfaceIds match {
+          case NonEmpty(_) => -\/(CannotQueryBothTemplateIdsAndInterfaceIds)
+          case _ => \/-(ByTemplateIds(templateIds))
+        }
+      case _ =>
+        interfaceIds match {
+          case NonEmpty(Singleton(interfaceId)) => \/-(ByInterfaceId(interfaceId))
+          case NonEmpty(_) => -\/(CannotQueryManyInterfaceIds)
+          case _ => -\/(CannotBeEmpty)
+        }
```

Comment:
Can you fix the typo in the error message a little further down:

```
final case object CannotQueryBothTemplateIdsAndInterfaceIds
      extends Unsupported("Cannot resolve any template ID from request")
```

Reply:
The original message: 
```
Cannot query both templates ID's and interface ID's
```

Thanks

Reply:
ad3e455a371f4393f419dd25f3e87ef8ba36f48d